### PR TITLE
Move slug validation to DeploymentManager

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  override def additionalPropertiesConfig(config: Config): Map[String, AdditionalPropertyConfig] = forMode(config)(
+    _ => Map.empty,
+    _ => RequestResponseOpenApiSettings.additionalPropertiesConfig
+  )
+
+  override def additionalValidators(config: Config): List[CustomProcessValidator] = forMode(config)(
+    _ => Nil,
+    config => List(new LiteScenarioValidator(config.nussknackerInstanceName))
```
Comment (arkadius) [inline engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: Do you see some problems in always returning this validator? I would like to keep `K8sDeploymentManager` as dynamic as possible - to distinguish between rr and streaming at last possible moment - when scenario is passed

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala [hunk 2]
```diff
@@ -20,12 +23,13 @@ class LiteScenarioValidator(nussknackerInstanceName: Option[String]) {
...
   }
 
-  private[manager] def validateRequestResponse(scenarioName: ProcessName, rrMetaData: RequestResponseMetaData): Validated[Throwable, Unit] = {
+  private[manager] def validateRequestResponse(scenarioName: ProcessName, rrMetaData: RequestResponseMetaData): ValidatedNel[ProcessCompilationError, Unit] = {
     val slug = RequestResponseSlugUtils.determineSlug(scenarioName, rrMetaData, nussknackerInstanceName)
     // We don't sanitize / validate against url because k8s object names are more restrictively validated than urls, see https://datatracker.ietf.org/doc/html/rfc3986
     val withoutSanitization = ServicePreparer.serviceNameWithoutSanitization(nussknackerInstanceName, slug)
     val withSanitization = ServicePreparer.serviceName(nussknackerInstanceName, slug)
-    Validated.cond(withSanitization == withoutSanitization, (), IllegalRequestResponseSlug(nussknackerInstanceName, slug))
+    Validated.cond(withSanitization == withoutSanitization, (), NonEmptyList.of(SpecificDataValidationError("slug", "Allowed characters include lowercase letters, digits, hyphen, " +
```
Comment (arkadius) [inline engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala:31]: What about name length restriction?

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala (hunk 1 above)
Comment (arkadius) [inline engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: What about validatePreparedService? IMO  we should run both validations:
- slug acceptable characters and length restrictions
- slug uniqueness
at once dynamically during validation of scenario.

Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: We check metadata type anyway, so this `forMode` is in fact not really needed. I think we have bigger problem of this kind with additionalPropertiesConfig and typeSpecificInitialData already...

Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: I would rather validate uniqueness only during deploy. We need access to K8s then, and it's more along the lines "transient errors (like duplication)" are done during deploy...

Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: On the other hand it is very easy to duplicate slug - let's say someone do export -> import of scenario. We validate scenario name, things with slug looks the same for me. Maybe we could do validation only on designer side by some query on DB :thinking: or some short living cache for this info from k8s. WDYT?

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala (hunk 2 above)
Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/LiteScenarioValidator.scala:32]: :+1: added

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala (hunk 1 above)
Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: We could... but I'd leave it for separate PR not to block the API

Comment (arkadius) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala:47]: ok

**Review by arkadius — APPROVED** [review]

File: engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala (hunk 1 above)
Comment (mproch) [reply engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala]: replaced generic validator with RequestResponseScenarioValidator

**Pull request merged**